google-cloud-pubsub/lib/google/cloud/pubsub/subscriber/async_pusher.rb

```diff
@@ -102,31 +102,18 @@ def delay deadline, ack_ids
 
           def stop
             synchronize do
-              break if @stopped
-
               @stopped = true
-              push_batch_request!
-              @cond.signal
-            end
 
-            self
-          end
-
-          def wait!
-            synchronize do
-              @background_thread.join if @background_thread
+              if @background_thread
+                @background_thread.kill
+              else
+                @background_thread = :stopped
```

Comment:
Sorry, I just don't follow. Why is  `@background_thread` assigned this symbol here?

Reply:
Oops, that conditional was left in from when I was diagnosing the deadlocks. I've removed it. Great catch!